src/texttospeechbar.cpp

```diff
@@ -82,14 +82,29 @@ void TextToSpeechBar::resetVoiceComboBox()
 
     m_voices = m_speech.availableVoices();
     const QVoice currentVoice = m_speech.voice();
+    const QString savedVoiceName = KiwixApp::instance()->getSavedVoiceName(m_speech.locale().name());
     for (const auto& voice : m_voices)
     {
         ui->voiceComboBox->addItem(QString("%1 - %2 - %3").arg(voice.name())
                           .arg(QVoice::genderName(voice.gender()))
                           .arg(QVoice::ageName(voice.age())));
-        if (voice.name() == currentVoice.name())
-            ui->voiceComboBox->setCurrentIndex(ui->voiceComboBox->count() - 1);
     }
+
+    /* We either stay with default voices or matches with the saved voice. */
+    int voiceIndex = 0;
+    for (int i = 0; i < ui->voiceComboBox->count(); i++)
+    {
+        if (m_voices[i].name() == currentVoice.name())
+            voiceIndex = i;
+        if (m_voices[i].name() == savedVoiceName)
+        {
+            voiceIndex = i;
+            break;
+        }
+    }
```

Comment:
Please introduce a function `int TextToSpeechBar::getVoiceIndex(const QString& voiceName) const` and use it by trying to first obtain the saved voice, or the current voice as a fallback.

Reply:
Done